linera-chain/src/chain.rs

```diff
@@ -557,80 +558,39 @@ where
             origin,
             bundle.height,
         );
-        // Process immediate messages and create inbox events.
-        let mut events = Vec::new();
-        for (index, outgoing_message) in bundle.messages {
-            ensure!(
```

Comment:
Is it safe to remove this check?

Reply:
`PostedMessage` doesn't contain the destination, unlike `OutgoingMessage`. That was duplicated information anyway: It was always the same as the bundle's destination, so this error was unreachable.

(If we assume that the machine sending that message wasn't faulty. But if it was, we couldn't rely on the destination here either.)